src/mmrelay/meshtastic_utils.py

```
@@ -441,14 +441,14 @@ async def _scan() -> bool:
-                except TypeError:
-                    return (
-                        await asyncio.wait_for(
-                            find_device(ble_address), timeout=timeout
-                        )
-                        is not None
+                    coro: Coroutine[Any, Any, Any] = cast(
+                        Coroutine[Any, Any, Any],
+                        find_device(ble_address, timeout=timeout),
                     )
+                    result = await coro
+                    return result is not None
```

Ruff (0.14.13)

Consider moving this statement to an `else` block

(TRY300)